### PR TITLE
Add _routing to SQL includes list (#277)

### PR DESCRIPTION
**Pull request by acarbonetto**
### Description
Add the `_routing` reserved word as a metafield accessible as an identifier from any opensearch index. 

Example: 
```
select _id, _index, _routing, int0 from calcs limit 5
```

Example response:
```
        [
            "1",
            "calcs",
            "[xoyIAOdqQsS74x2yKSgiEQ][calcs][0]",
            1
        ],
```
 
### Issues Resolved
Partially resolves: [sql#1478](https://github.com/opensearch-project/sql/issues/1478)
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Yury-Fridlyand — DISMISSED** [review]
I think we need an IT to prove that filtering by `_routing` works fine, for example:
```sql
select *, _id from <index> where _routing = 'a'
select *, _id from <index> where _routing = 'b'
```
The test should propagate the data properly into different shards and get their names using REST API.

**Comment by acarbonetto** [comment]
> I think we need an IT to prove that filtering by `_routing` works fine, for example:
> 
> ```sql
> select *, _id from <index> where _routing = 'a'
> select *, _id from <index> where _routing = 'b'
> ```
> 
> The test should propagate the data properly into different shards and get their names using REST API.

Right now routing shard names are randomly assigned, so I cannot filter by shard name.  Yet.  
In the following PR, I will be able to assign documents to specific shards, and will be able to filter by routing shard using exactly what you were suggesting. 

Note: this is also why a doctest with _routing doesn't work, because the documents are assigned to random shards.

**Comment by Yury-Fridlyand** [comment]
Yes, but you can get shard names with REST API or CAT API

**Review by MitchellGale — APPROVED** [review]

File: integ-test/src/test/java/org/opensearch/sql/sql/IdentifierIT.java [hunk 1]
```diff
@@ -99,6 +101,78 @@ public void testMetafieldIdentifierTest() throws IOException {
...
+    verifySchema(result,
+        schema("_id", null, "keyword"),
+        schema("_index", null, "keyword"),
+        schema("_routing", null, "keyword"));
+    assertTrue(result.getJSONArray("schema").length() == 3);
+
+    var datarows = result.getJSONArray("datarows");
+    assertEquals(1, datarows.length());
+
+    // note that _routing in the SELECT clause returns the shard
```
Comment (Yury-Fridlyand) [inline integ-test/src/test/java/org/opensearch/sql/sql/IdentifierIT.java:172]: Maybe add this phrase to docs?

**Review by Yury-Fridlyand — DISMISSED** [review]

File: integ-test/src/test/java/org/opensearch/sql/sql/IdentifierIT.java [hunk 2]
```diff
@@ -99,6 +101,78 @@ public void testMetafieldIdentifierTest() throws IOException {
...
+    assertTrue(datarows.getJSONArray(0).getString(2).contains("[" + index + "]"));
```
Comment (dai-chen) [inline integ-test/src/test/java/org/opensearch/sql/sql/IdentifierIT.java:173]: Should we assert doc id is "test4"?

**Review by dai-chen — APPROVED** [review]
Thanks for the changes!

File: integ-test/src/test/java/org/opensearch/sql/sql/IdentifierIT.java (hunk 2 above)
Comment (acarbonetto) [reply integ-test/src/test/java/org/opensearch/sql/sql/IdentifierIT.java:174]: Updated

**Review by Yury-Fridlyand — APPROVED** [review]

**Review by MaxKsyunz — APPROVED** [review]

**Pull request merged**